### PR TITLE
Updates for support in Jenkinsfile

### PR DESCRIPTION
**Pull request by shawnallen85**
Previously, trying to use this plugin when referencing the global tool inside of a Jenkinsfile failed miserably.

I've made a few updates to get it working.

Example:

pipeline {
    agent any
    tools {
        dependencyCheck 'DependencyCheck5'
    }
    stages {
        stage('Example') {
            steps {
                sh 'dependency-check.sh --version'
            }
        }
    }
}

There are two issues that have been fixed with this.

First, is the Symbol annotation on DescriptorImpl. I think the "-" was main culprit. I updated the string value to be "dependencyCheck" (rather than "dependency-check").

Second, was not setting the PATH environment variable appropriately to reference the tool.

I also made a change to how it was determined if Jenkins was on a Windows node or not.

### CODE REVIEW

**Comment by ChristianCiach** [comment]
Does this PR also address the issue that the Tool is not available when building inside docker containers? Usually, when referencing tools using the `tools`-block of a declarative pipeline, the referenced tool gets mounted into the container. This currently doesn't work. Maybe this PR will fix this as well.

**Comment by shawnallen85** [comment]
I believe the answer to that is yes.

On my current Jenkins instance, I have the following Jenkinsfile. The Docker image this runs does *not* include dependency check.

```groovy
pipeline {
    agent {
        label 'linux && x86_64 && debian'
    }
    tools {
        dependencyCheck 'DependencyCheck5'
    }
    stages {
        stage('Hello!') {
            steps {
                sh 'dependency-check.sh --help'
            }
        }
    }
}
```

Output:
```
+ dependency-check.sh --help
usage: Dependency-Check Core [--advancedHelp] [--enableExperimental]
       [--exclude <pattern>] [-f <format>] [--failOnCVSS <score>] [-h]
       [--junitFailOnCVSS <score>] [-l <file>] [-n] [-o <path>]
       [--prettyPrint] [--project <name>] [-s <path>] [--suppression
       <file>] [-v]

Dependency-Check Core can be used to identify if there are any known CVE
vulnerabilities in libraries utilized by an application. Dependency-Check
Core will automatically update required data from the Internet, such as
the CVE and CPE data files from nvd.nist.gov.

    --advancedHelp              Print the advanced help message.
    --enableExperimental        Enables the experimental analyzers.
    --exclude <pattern>         Specify an exclusion pattern. This option
                                can be specified multiple times and it
                                accepts Ant style exclusions.
 -f,--format <format>           The report format (HTML, XML, CSV, JSON,
                                JUNIT, or ALL). The default is HTML.
                                Multiple format parameters can be
                                specified.
    --failOnCVSS <score>        Specifies if the build should be failed if
                                a CVSS score above a specified level is
                                identified. The default is 11; since the
                                CVSS scores are 0-10, by default the build
                                will never fail.
 -h,--help                      Print this message.
    --junitFailOnCVSS <score>   Specifies the CVSS score that is
                                considered a failure when generating the
                                junit report. The default is 0.
 -l,--log <file>                The file path to write verbose logging
                                information.
 -n,--noupdate                  Disables the automatic updating of the CPE
                                data.
 -o,--out <path>                The folder to write reports to. This
                                defaults to the current directory. It is
                                possible to set this to a specific file
                                name if the format argument is not set to
                                ALL.
    --prettyPrint               When specified the JSON and XML report
                                formats will be pretty printed.
    --project <name>            The name of the project being scanned.
 -s,--scan <path>               The path to scan - this option can be
                                specified multiple times. Ant style paths
                                are supported (e.g. 'path/**/*.jar'); if
                                using Ant style paths it is highly
                                recommended to quote the argument value.
    --suppression <file>        The file path to the suppression XML file.
                                This can be specified more then once to
                                utilize multiple suppression files
 -v,--version                   Print the version information.
```

**Comment by shawnallen85** [comment]
Anybody home?

**Comment by stevespringett** [comment]
Refer to https://github.com/jenkinsci/dependency-check-plugin/blob/master/README.md#seeking-new-maintainer

**Review by yjagdale — APPROVED** [review]

File: src/main/java/org/jenkinsci/plugins/DependencyCheck/tools/DependencyCheckInstallation.java [hunk 1]
```diff
@@ -61,36 +62,37 @@ public DependencyCheckInstallation forEnvironment(EnvVars environment) {
...
-            return home + File.separator + "bin" + File.separator + command;
-        }
+    private File getExeFile() {
+        String execName = (Functions.isWindows()) ? "dependency-check.bat" : "dependency-check.sh";
+        return new File(getHome(), execName);
     }
 
     @Extension
-    @Symbol("dependency-check")
+    @Symbol("dependencyCheck")
```
Comment (nfalco79) [inline src/main/java/org/jenkinsci/plugins/DependencyCheck/tools/DependencyCheckInstallation.java:95]: not backward compatible

File: src/main/java/org/jenkinsci/plugins/DependencyCheck/tools/DependencyCheckInstallation.java [hunk 2]
```diff
@@ -61,36 +62,37 @@ public DependencyCheckInstallation forEnvironment(EnvVars environment) {
...
     public DependencyCheckInstallation forNode(@Nonnull Node node, TaskListener log) throws IOException, InterruptedException {
         return new DependencyCheckInstallation(getName(), translateFor(node, log), getProperties().toList());
     }
 
-    public String getExecutable(@Nonnull Launcher launcher) throws IOException, InterruptedException {
-        final VirtualChannel channel = launcher.getChannel();
-        return channel == null ? null : channel.call(new FindExecutableCallable(getHome()));
+    @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
+    public String getExecutable(Launcher launcher) throws IOException, InterruptedException {
+        return launcher.getChannel().call(new MasterToSlaveCallable<String, IOException>() {
```
Comment (nfalco79) [inline src/main/java/org/jenkinsci/plugins/DependencyCheck/tools/DependencyCheckInstallation.java:78]: this implementation cause log on jenkins console becasuse of use of anynoums class
https://www.jenkins.io/doc/developer/security/remoting-callables/#minimal-safe-implementations

File: src/main/java/org/jenkinsci/plugins/DependencyCheck/tools/DependencyCheckInstallation.java [hunk 3]
```diff
@@ -61,36 +62,37 @@ public DependencyCheckInstallation forEnvironment(EnvVars environment) {
         return new DependencyCheckInstallation(getName(), environment.expand(getHome()), getProperties().toList());
     }
 
+    @Override
+    public void buildEnvVars(EnvVars env) {
+        if("".equals(env.get("PATH+DependencyCheck", "")))
```
Comment (nfalco79) [inline src/main/java/org/jenkinsci/plugins/DependencyCheck/tools/DependencyCheckInstallation.java:67]: it's not needed to check if exists, put into env directly

**Pull request closed without merging**